yugabyted-ui/apiserver/cmd/server/handlers/api_voyager.go

```diff
@@ -189,89 +248,248 @@ func getVoyagerMigrationsQueryFuture(log logger.Logger, conn *pgxpool.Pool,
+        // This is done since source db info is only in export rows, target db info in import rows.
+        // Info from import rows will overwrite info from export rows.
+        rowStruct := models.VoyagerMigrationDetails{}
+        if allVoyagerMigration.exportMigrationPhase.Status == pgtype.Present &&
+            allVoyagerMigration.exportInvocationSeq.Status == pgtype.Present {
+            err := updateRowStruct(log, conn, &rowStruct, allVoyagerMigration.migrationUuid,
+                allVoyagerMigration.exportMigrationPhase.Int,
+                allVoyagerMigration.exportInvocationSeq.Int)
+            if err != nil {
+                log.Errorf("[%s] Error while querying for export phase migration details",
+                    LOGGER_FILE_NAME)
+            }
+        }
+        if allVoyagerMigration.importMigrationPhase.Status == pgtype.Present &&
+            allVoyagerMigration.importInvocationSeq.Status == pgtype.Present {
+            err := updateRowStruct(log, conn, &rowStruct, allVoyagerMigration.migrationUuid,
+                allVoyagerMigration.importMigrationPhase.Int,
+                allVoyagerMigration.importInvocationSeq.Int)
+            if err != nil {
+                log.Errorf("[%s] Error while querying for import phase migration details",
+                    LOGGER_FILE_NAME)
+            }
+        }
+        if allVoyagerMigration.lowestMigrationPhase.Status == pgtype.Present &&
+            allVoyagerMigration.lowestInvocationSeq.Status == pgtype.Present {
+            err := updateRowStructStartTimestamp(log, conn, &rowStruct,
+                allVoyagerMigration.migrationUuid,
+                allVoyagerMigration.lowestMigrationPhase.Int,
+                allVoyagerMigration.lowestInvocationSeq.Int)
+            if err != nil {
+                log.Errorf("[%s] Error while querying for migration start timestamp",
+                    LOGGER_FILE_NAME)
+            }
+        }
+
+        // Get complexity from assessment if it exists.
+        rowStruct.Complexity = "N/A"
+        assessment := <-assessmentFuture
+        if assessment.Error != nil {
+            log.Errorf("[%s] Error getting migration assessment",
+                LOGGER_FILE_NAME)
         } else {
-            migrationPhaseToBeUsed = allVoygaerMigration.migrationPhase
-            migrationInvocationSqToBeUsed = allVoygaerMigration.invocationSeq
+            rowStruct.Complexity = assessment.Report.MigrationComplexity
         }
-        voyagerDetailsrows, err := conn.Query(context.Background(),
-            RETRIEVE_VOYGAGER_MIGRATION_DETAILS,
-            allVoygaerMigration.migrationUuid, migrationPhaseToBeUsed,
-            migrationInvocationSqToBeUsed)
+        voyagerMigrationsResponse.Migrations = append(
+            voyagerMigrationsResponse.Migrations, rowStruct)
+    }
+    future <- voyagerMigrationsResponse
+}
+
+func updateRowStruct(log logger.Logger, conn *pgxpool.Pool,
```

Comment:
Can we have a better name for the function to describe the outcome of invoking this function?